packages/type-utils/src/typeFlagUtils.ts

```diff
@@ -1,20 +1,27 @@
 import { unionTypeParts } from 'tsutils';
 import * as ts from 'typescript';
 
+const ANY_OR_UNKNOWN = ts.TypeFlags.Any | ts.TypeFlags.Unknown;
+
 /**
- * Gets all of the type flags in a type, iterating through unions automatically
+ * Gets all of the type flags in a type, iterating through unions automatically.
  */
 export function getTypeFlags(type: ts.Type): ts.TypeFlags {
-  let flags: ts.TypeFlags = 0;
+  let flags = 0 as ts.TypeFlags;
```

Comment:
I wonder if it would be worth adding an option to allowlist an enum name pattern as a knowingly unsafe flags enum the rule should ignore?

Reply:
isn't `flags: ts.TypeFlags = 0` since it type checks that 0 can fit inside fo the `ts.TypeFlags` type? whereas the `as` assertion disables the type checker completely

Reply:
nope - TS doesn't do ***any*** validation of number enums. It's why they're so *completely* unsafe.

`0` isn't even a valid `ts.TypeFlags` value!
https://github.com/microsoft/TypeScript/blob/ff919e3b76c096547a4a33d7632102382eb025cd/src/compiler/types.ts#L5999-L6089

Reply:
This should be possible with the new rule. So I am unsure if an allowlist is needed.
```ts
let flags: ts.TypeFlags | 0 = 0;

Reply:
> TS doesn't do any validation of number enums.

ok, i know that in general, TS doesn't do any validation of number enums.
but *in general* the pattern I describe is safer typescript code, and if number enums get future improvements, then the pattern I describe would be better than using `as`.

furthermore, if 0 isn't a valid `ts.TypeFlags` value, then it should probably be expressed as Holger says in order to demonstrate that within the code

Reply:
I'd say let's keep this as a followup if people request it. It can use the format from #6017.